src/lib/libast/sfio/sfrd.c

```diff
@@ -275,20 +276,11 @@ ssize_t sfrd(Sfio_t *f, void *buf, size_t n, Sfdisc_t *disc) {
 
     do_except:
         if (local) SETLOCAL(f);
-        switch (_sfexcept(f, SF_READ, (ssize_t)r, dc)) {
-            case SF_ECONT:
```

Comment:
This line is not covered by tests. While this change looks correct on a quick look, it would be great to have a test for it.